src/app/layout/qgslayoutapputils.cpp

```diff
@@ -195,6 +195,16 @@ void QgsLayoutAppUtils::registerGuiForKnownItemTypes()
     // try to find a good map to link the legend with by default
     legend->setLinkedMap( findSensibleDefaultLinkedMapItem( legend ) );
 
+    if ( QApplication::isRightToLeft() )
+    {
+      // for right-to-left locales, use an appropriate default layout
```

Comment:
Bonus point if this great logic is applied to label items too (no need to part of this PR ;) )